spot-contracts/contracts/PerpetualTranche.sol

```diff
@@ -959,13 +922,28 @@ contract PerpetualTranche is
             return r;
         }
 
+        //-----------------------------------------------------------------------------
+        // Basic rollover with fees:
+        // (1 +/- f) . (stdTrancheInAmt . trancheInPrice) = (stdTrancheOutAmt . trancheOutPrice)
+        //-----------------------------------------------------------------------------
+
+        // Given the amount of tranches In, we compute the amount of tranches out
         r.trancheInAmt = trancheInAmtAvailable;
         uint256 stdTrancheInAmt = _toStdTrancheAmt(trancheInAmtAvailable, trancheInDiscount);
-
-        // Basic rollover:
-        // (stdTrancheInAmt . trancheInPrice) = (stdTrancheOutAmt . trancheOutPrice)
         uint256 stdTrancheOutAmt = stdTrancheInAmt.mulDiv(trancheInPrice, trancheOutPrice);
-        r.trancheOutAmt = _fromStdTrancheAmt(stdTrancheOutAmt, trancheOutDiscount);
+
+        // A postive fee percentage implies that perp charges rotators by
+        // accepting tranchesIn at a discount, ie) lesser tranches out.
+        if (feePerc > 0) {
```

Comment:
I guess it's worth asking if these two conditions could be collapsed into one.

`feePerc` is already a signed number, so could we have one condition that simply does `HUNDRED_PERC + feePerc`?

But I'd keep the comments, those are very helpful.

Reply:
Maybe it just collapses into line 933

However, if we round up in one case, and down in the other, then we should keep them separate though